Parse/PFObject.m

```diff
@@ -350,29 +350,31 @@ + (BOOL)canBeSerializedAsValue:(id)value
 // @param saved A set of objects that we can assume will have been saved.
 // @param error The reason why it can't be serialized.
 - (BOOL)canBeSerializedAfterSaving:(NSMutableArray *)saved withCurrentUser:(PFUser *)user error:(NSError **)error {
+    NSDictionary *dictionaryRepresentationCopy;
     @synchronized (lock) {
-        // This method is only used for batching sets of objects for saveAll
-        // and when saving children automatically. Since it's only used to
-        // determine whether or not save should be called on them, it only
-        // needs to examine their current values, so we use estimatedData.
-        if (![[self class] canBeSerializedAsValue:_estimatedData.dictionaryRepresentation
-                                      afterSaving:saved
-                                            error:error]) {
-            return NO;
-        }
-
-        if ([self isDataAvailableForKey:@"ACL"] &&
-            [[self ACLWithoutCopying] hasUnresolvedUser] &&
-            ![saved containsObject:user]) {
-            if (error) {
-                *error = [PFErrorUtilities errorWithCode:kPFErrorInvalidACL
-                                                 message:@"User associated with ACL must be signed up."];
-            }
-            return NO;
+        dictionaryRepresentationCopy = _estimatedData.dictionaryRepresentation; // This copies the dictionary representation
```

Comment:
this actually doesn't copy anything, just assigns the pointer, unless you call `[_estimatedData.dictionaryRepresentation copy]` on it.